libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -840,6 +846,15 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         CacheHandleWrapper.requestRoadGraphDataUpdate(navigator.cache, callback)
     }
 
+    // call before starting session - to docs
+    fun registerHistoryRecordingStateChangeObserver(observer: HistoryRecordingStateChangeObserver) {
+        historyRecordingStateHandler.registerHistoryRecordingStateChangeObserver(observer)
+    }
+
+    fun unregisterHistoryRecordingStateChangeObserver(observer: HistoryRecordingStateChangeObserver) {
+        historyRecordingStateHandler.unregisterHistoryRecordingStateChangeObserver(observer)
+    }
```

Comment:
These should rather be extensions on `MapboxNavigation` exposed from `internal` package, or at least tagged with experimental.

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/6074